### PR TITLE
Little Fix for AI Related Gamemodes(aitdm and fw)

### PR DESCRIPTION
**Pull request by DBmaoha**
Changes:
- General
    - Add `AiGameModes_SetNPCWeapons( string npcClass, array<string> weapons )` to replace `AiGameModes_SetGruntWeapons()` and `AiGameModes_SetSpectreWeapons()`, which can handle more npc weapon settings
    - Rework `AiGameModes_SpawnDropPod()`, it now spawns npcs after droppod actually lands, so npcs won't show their minimap icon and name in the sky, same as vanilla does
    - Rename `Reaper_Spawnpoint()` to `HotDrop_Spawnpoint()`, an option to add dangerous area for the spawnpoint, so other npcs will try to avoid their crush damage
    
- AITdm( Attrition )
    - Change to use `AiGameModes_SetNPCWeapons()` for npc weapon setup
    - Stalkers now also uses Mastiff and Volt, same as vanilla
    - Add settings for each levels, for modders can change them easily,  functions:
        - `AITdm_SetSquadsPerTeam( int count )`
        - `AITdm_SetReapersPerTeam( int count )`
        - `AITdm_SetLevelSpectres( int level )`
        - `AITdm_SetLevelStalkers( int level )`
        - `AITdm_SetLevelReapers( int level )`
    - Rework `SquadHandler()`, npcs will wait for enemy spawn and try to move together with their squadmates
- FW ( Frontier War )
    - Change to use `AiGameModes_SetNPCWeapons()` for npc weapon setup

### CODE REVIEW

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_ai_gamemodes.gnut [hunk 1]
```diff
@@ -15,25 +14,21 @@ const INTRO_DROPSHIP_CUTOFF = 2000
...
 {
-	array< string > gruntWeapons = [ "mp_weapon_rspn101" ]
-	array< string > spectreWeapons = [ "mp_weapon_hemlok_smg" ]
+	table< string, array<string> > npcWeaponsTable // npcs have their default weapon in aisettings file
 } file
 
 void function AiGameModes_Init()
 {
-
+
```
Comment (F1F7Y) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_ai_gamemodes.gnut:22]: Whitespace change

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
+global function AITdm_SetReapersPerTeam
+global function AITdm_SetLevelSpectres
+global function AITdm_SetLevelStalkers
+global function AITdm_SetLevelReapers
 
 struct
 {
 	// Due to team based escalation everything is an array
-	array< int > levels = [ LEVEL_SPECTRES, LEVEL_SPECTRES ]
+	array< int > levels = [] //[ LEVEL_SPECTRES, LEVEL_SPECTRES ] // since we added modification should leave these to the start of spawner
```
Comment (F1F7Y) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut:23]: No need to keep the `//[ LEVEL_SPECTRES, LEVEL_SPECTRES ]`

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut [hunk 3]
```diff
@@ -74,11 +117,11 @@ void function HandleScoreEvent( entity victim, entity attacker, var damageInfo )
 	// Basic checks
 	if ( victim == attacker || !( attacker.IsPlayer() || attacker.IsTitan() ) || GetGameState() != eGameState.Playing )
 		return
-	
+
```
Comment (F1F7Y) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut:120]: Again, unnecessary whitespace change

**Review by F1F7Y — CHANGES_REQUESTED** [review]
Looks good, just some formatting. Making sure your editor doesnt strip line-ending whitespace characters just makes it easier to git blame

**Review by F1F7Y — APPROVED** [review]
Code looks good, will test on friday

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut [hunk 4]
```diff
@@ -1,22 +1,36 @@
...
+global function AITdm_SetReapersPerTeam
+global function AITdm_SetLevelSpectres
+global function AITdm_SetLevelStalkers
+global function AITdm_SetLevelReapers
 
 struct
 {
 	// Due to team based escalation everything is an array
-	array< int > levels = [ LEVEL_SPECTRES, LEVEL_SPECTRES ]
+	array< int > levels = [] // since we added modification should init starter level int Spawner()
```
Comment (GeckoEidechse) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut:23]: I'm not quite sure what that comment means... Anyone can ELI5? ^^"

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut [hunk 5]
```diff
@@ -1,22 +1,36 @@
 untyped
 global function GamemodeAITdm_Init
 
-const SQUADS_PER_TEAM = 3
+// these are now default settings
+const int SQUADS_PER_TEAM = 4 // was 3
```
Comment (GeckoEidechse) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut:5]: Wait, if I'm reading this correctly, it previously only spawned 3 grunts per pod/dropship? o_O

Comment (GeckoEidechse) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut:5]: No need to save old value in comment. That's what commit history and `git blame` is for :P
```suggestion
const int SQUADS_PER_TEAM = 4
```

**Review by GeckoEidechse — COMMENTED** [review]
Just some stuff I found while quickly skimming over diff
(mostly just [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding))

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut (hunk 5 above)
Comment (F1F7Y) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut:5]: `SQUADS_PER_TEAM` determines how many AI squads there are per team, each squad has 4 npcs

Comment (F1F7Y) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut:5]: Correct

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut (hunk 4 above)
Comment (F1F7Y) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_aitdm.nut]: The array is initilazed later so that mods can modifiy the values using the new `AITdm_SetLevel*` functions

**Review by F1F7Y — APPROVED** [review]
Code looks good, works in game ( tested only on a 1 player match )

**Review by uniboi — APPROVED** [review]
Code looks good and I didn't notice anything odd in a quick local match
